doc/building.md

```diff
@@ -4,37 +4,53 @@ This guide documents the Onyx build process. For example usage of this building
 
-First, you'll need to download the [minimal sysroot](https://storage.googleapis.com/onyx-stuffs/minimal-sysroot.tar.zst).
-This allows you to have the toolchain build against a sane libc.
-Be sure to export `SYSROOT=$PWD/sysroot` and `ONYX_ARCH=x86_64` (right now, only x86_64 is supported).
+You need the following packages and programs: mtools, xorriso, libfl2, ninja, gn, grub-mkrescue.
+Also you need development files for MPC, MPFR and GMP and texinfo package if you are going to build toolchains.
 
-## Step 1: Get the toolchains
+## Step 1: Fetch submodules
 
-If you so desire, you can find prebuilts for Linux and macOS in the project's github action artifacts,
-under `build-toolchain-${os}` or `build-toolchain-llvm-${os}`. Note that macOS LLVM builds are broken.
+```
+git submodule init
+git submodule update --recursive
+```
+
+## Step 2: Get minimal sysroot (skip if using prebuilt toolchain)
+
+First, you'll need to download the minimal sysroot available in project's github actions artifacts.
+This allows you to have the toolchain build against a sane libc. Uncompress it into the project root.
+
+
+## Step 3: Prepare build environment
 
-You can find two scripts, `build_gcc.sh` and `build_llvm.sh`, under `toolchains/scripts`. These build a
-GCC and LLVM toolchain for Onyx, respectively. Use these and install the toolchains to a known location.
-Note: $GCC_TOOLCHAIN/bin needs to be on the system's $PATH. LLVM has no such requirement.
+1. `export ONYX_ARCH=x86_64`
+2. Install appropriate kernel config under `kernel/kernel.config`. There are pre-existing examples in
+`kernel/configs`, these are fine choices.
+3. Run `scripts/setup_build.sh`
+
+## Step 4: Build or obtain the toolchains
+
+### Obtaining prebuilt toolchains
+
+These are available for Linux and macOS in the project's github action artifacts,
+under `build-toolchain-${os}` or `build-toolchain-llvm-${os}`. Note that macOS LLVM builds are broken.
 
-## Step 2: Set up the kernel config
+### Building toolchains
 
-The kernel build requires a valid `kernel.config` under kernel/. You can look at pre-existing examples
-(.example and .minimal) and lay the kernel out yourself; kernel.config.example is a fine choice.
+`toolchains/scripts/build_toolchain.sh` will do it for you. You need to choose staging and target
+directory toolchain. Target toolchain directory further will be referred as `$TOOLCHAIN_TARGET_DIR`.
 
-After this point, it's required that SYSROOT, ONYX_ARCH are set and that, if using LLVM, CLANG_PATH points to the
-installation path of the LLVM toolchain.
+### Adding toolchain paths to environment
 
-## Step 3: Set up the build
+If using **gcc**: add `$TOOLCHAIN_TARGET_DIR/bin` to `$PATH`.
 
-This step only needs to be done once. Run `./scripts/setup_build.sh`.
+If using **LLVM**: `export CLANG_PATH=$TOOLCHAIN_TARGET_DIR/bin`
 
-## Step 4: Build
+## Step 5: Build
 
 Do `make -j <nproc> iso` where nproc is the number of threads you want the build to use.
```

Comment:
Since the system atm is mostly much more usable in a live fashion, maybe adjust this for make liveiso